### PR TITLE
Add attachment

### PR DESCRIPTION
**Pull request by mjagelka**

### CODE REVIEW

File: tests/test_servicenow.py [hunk 1]
```diff
@@ -316,5 +316,46 @@ def test_remove_cc_unexpected_response(self, mock_session):
...
+
+    @patch('builtins.open')
+    @patch.object(servicenow.ServiceNowTicket, '_create_requests_session')
+    @patch('servicenow.ServiceNowTicket._verify_project', mock_verify_project)
+    @patch('servicenow.ServiceNowTicket.get_ticket_content', mock_get_ticket_content)
+    def test_add_attachment_unexpected_response(self, mock_session, mock_open):
+        mock_session.return_value = FakeSession(status_code=404)
+        ticket = servicenow.ServiceNowTicket(TEST_URL, TABLE, ticket_id=TICKET_ID)
+        t = ticket.add_attachment('file_name')
+        self.assertEqual(t.error_message, '')
```
Comment (dmranck) [inline tests/test_servicenow.py]: This unit test passed, by why is t.error_message equal to an empty string in this case? Shouldn't it be str(e), which is RequestException?

Comment (mjagelka) [reply tests/test_servicenow.py]: This was quite intuitive for me, so I am not sure if my reasoning is correct. str(e) returns the result of RequestException method `str`, which is an empty string if we raise flat RequestException without parameters, as it is at l.63 in test_servicenow.py. Take a look at similar example: https://stackoverflow.com/questions/17178186/empty-error-message-in-python

**Comment by mjagelka** [comment]
The ServiceNow.rst file was updated.

File: tests/test_servicenow.py (hunk 1 above)
Comment (dmranck) [reply tests/test_servicenow.py]: Ok, this makes sense. Thanks. :)

**Pull request merged**